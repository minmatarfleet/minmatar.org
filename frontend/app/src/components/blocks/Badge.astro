---
import type { BadgeColors } from '@dtypes/LayoutComponents';

interface Props {
    number: number;
    color?: BadgeColors;
    [propName: string]: any;
}

const {
    number,
    color = 'fleet-red',
    ...attributes
} = Astro.props;

const badge_color = `var(--${color})`;
---

<span class="[ badge ]" {...attributes}>{number}</span>

<style lang="scss" define:vars={{ badge_color }}>
    .badge {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 1.2rem;
        height: 1.2rem;
        border-radius: 50%;
        font-size: var(--step--2);
        font-family: var(--badge-font);
        color: var(--highlight);
        background-color: var(--badge_color);
    }
</style>