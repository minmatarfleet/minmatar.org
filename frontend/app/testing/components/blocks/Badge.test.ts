import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import Badge from "@components/blocks/Badge.astro";

test("Badge defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(Badge, {});

  expect(result).toMatchSnapshot();
});
