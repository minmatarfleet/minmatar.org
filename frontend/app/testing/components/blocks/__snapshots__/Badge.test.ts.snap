// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge defaults 1`] = `"<div class="fixed-fluid [ items-center ]" data-astro-cid-etghgjam="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">   <picture data-astro-cid-etghgjam>  </picture>  <div class="flex-block gap-0" data-astro-cid-etghgjam="true" data-mod style="--flex_block_gap: 0;">  <span data-astro-cid-etghgjam="true" class="min-w-[200px] text-[var(--highlight)]"> </span>     </div>   </div>   "`;
